makeflow/src/makeflow_mpi_submitter.c

```diff
@@ -0,0 +1,524 @@
+        binary = "mpi_worker";
+        if (jx_lookup_string(options, "workqueue-arguments") != NULL) {
+            workqueue_options = (char*) jx_lookup_string(options, "workqueue-arguments");
+        }
+    }
+    
+    if(jx_lookup_integer(options,"memory") != 0){
+        workqueue_options = string_format("--memory=%i %s",(int)jx_lookup_integer(options,"memory"),workqueue_options);
+    }
+    if(jx_lookup_string(options,"disk")!=NULL){
+        workqueue_options = string_format("--disk=%s %s",jx_lookup_string(options,"disk"),workqueue_options);
+    }
+    if(jx_lookup_string(options,"disk-location")!=NULL){
+        workqueue_options = string_format("--workdir=%s %s",jx_lookup_string(options,"disk-location"),workqueue_options);
+    }
+    
+    fprintf(fout, "#!/bin/sh\n\n");
+
+    if (jx_lookup_string(options, "email") != NULL) {
+        fprintf(fout, "#PBS -M %s\n", jx_lookup_string(options, "email")); //email
+        fprintf(fout, "#PBS -m abe\n");
+    }
+    fprintf(fout, "#PBS -N %s\n", generate_job_name()); //job name
+    fprintf(fout, "#PBS -j oe\n"); //join stdout and stderr
+    fprintf(fout, "#PBS -k o\n"); //keeps output
+    fprintf(fout, "#PBS -V\n"); //pass on submitter's environment
+    
+    //do all the fun nodes and processes and stuff....
+    fprintf(fout, "#PBS -l nodes=%i\n",(int)jx_lookup_integer(options,"slots"));
+    fprintf(fout, "#PBS -l ppn=1\n"); //we're going to have to say 1, and if cores-per-worker is set, then we can send it, otherwise, set --cores=0 in the worker settings
+    if(jx_lookup_integer(options,"cores-per-worker") != 0){
+        workqueue_options = string_format("--cores=%i %s",(int)jx_lookup_integer(options,"cores-per-worker"),workqueue_options);
+        if (strstr("makeflow -T mpi", binary)) {
+            makeflow_options = string_format("--mpi-cores=%i %s", (int) jx_lookup_integer(options, "cores-per-worker"), makeflow_options);
+        }
+    }else{
+        workqueue_options = string_format("--cores=0 %s",workqueue_options);
+    }
+    //if(jx_lookup_integer(options,"memory")!=0){
+    //    fprintf(fout,"#PBS -l pmem=%imb\n",jx_lookup_integer(options,"memory"));
+    //}
+    
+    if (jx_lookup_string(options, "mpi-module") != NULL) {
+        fprintf(fout, "module load %s\n", jx_lookup_string(options, "mpi-module")); //assume this works
+    }
+    
+    if(strlen(workqueue_options)>0 && !strstr("mpi_worker",binary)){
+        workqueue_options = string_format("-q \"%s\"",workqueue_options);
+    }
+    if(strlen(makeflow_options)>0 && !(strstr("mpi_worker",binary) || strstr("makeflow -T mpi",binary))){
+        makeflow_options = string_format("-m \"%s\"",makeflow_options);
+    }
+    if(strstr("makeflow -T mpi",binary)){
+        workqueue_options = "";
+    }
+    fprintf(fout, "mpirun -npernode 1 -machinefile $PBS_NODEFILE %s %s %s\n", binary, makeflow_options, workqueue_options);
+    
+    
+    
+    fclose(fout);
+}
+
+enum submitter_types {
+    slurm = 1234,
```

Comment:
Is 1234 significant?

Reply:
Please remove the '= 1234', so people looking at the code in the future don't get confused.

Reply:
Done